--- bcachefs-tools-1.4.0/Makefile.orig	2023-12-24 01:39:01.000000000 +0100
+++ bcachefs-tools-1.4.0/Makefile	2024-01-12 18:28:15.566397728 +0100
@@ -58,6 +58,13 @@
 CARGO_MANIFEST=--manifest-path rust-src/Cargo.toml
 
 CARGO_BUILD_ARGS=--$(CARGO_PROFILE)
+
+ifdef CARGO_TARGET
+	CARGO_BUILD_ARGS += --target $(CARGO_TARGET)
+	CARGO_TARGET_DIR := target/$(CARGO_TARGET)
+else
+	CARGO_TARGET_DIR := target
+endif
 CARGO_BUILD=$(CARGO) build $(CARGO_BUILD_ARGS) $(CARGO_MANIFEST)
 
 CARGO_CLEAN=$(CARGO) clean $(CARGO_CLEAN_ARGS) $(CARGO_MANIFEST)
@@ -174,7 +181,7 @@
 BCACHEFS_DEPS=libbcachefs.a
 
 ifndef NO_RUST
-	BCACHEFS_DEPS+=rust-src/target/release/libbcachefs_rust.a
+	BCACHEFS_DEPS+=rust-src/$(CARGO_TARGET_DIR)/release/libbcachefs_rust.a
 else
 	CFLAGS+=-DBCACHEFS_NO_RUST
 endif
@@ -188,7 +195,7 @@
 	$(Q)ar -rc $@ $+
 
 RUST_SRCS:=$(shell find rust-src/src rust-src/bch_bindgen/src -type f -iname '*.rs')
-rust-src/target/release/libbcachefs_rust.a: $(RUST_SRCS)
+rust-src/$(CARGO_TARGET_DIR)/release/libbcachefs_rust.a: $(RUST_SRCS)
 	$(CARGO_BUILD)
 
 tests/test_helper: $(filter ./tests/%.o, $(OBJS))
