--- bcachefs-tools-1.7.0/Makefile.orig	2024-05-23 22:32:41.238838991 +0200
+++ bcachefs-tools-1.7.0/Makefile	2024-05-24 16:34:59.100373730 +0200
@@ -59,6 +59,13 @@ CARGO_PROFILE=release
 # CARGO_PROFILE=debug
 
 CARGO_BUILD_ARGS=--$(CARGO_PROFILE)
+
+ifdef CARGO_TARGET
+	CARGO_BUILD_ARGS += --target $(CARGO_TARGET)
+	CARGO_TARGET_DIR := target/$(CARGO_TARGET)
+else
+	CARGO_TARGET_DIR := target
+endif
 CARGO_BUILD=$(CARGO) build $(CARGO_BUILD_ARGS)
 
 CARGO_CLEAN=$(CARGO) clean $(CARGO_CLEAN_ARGS)
@@ -201,7 +208,7 @@ cmd_version.o : .version
 install: INITRAMFS_HOOK=$(INITRAMFS_DIR)/hooks/bcachefs
 install: INITRAMFS_SCRIPT=$(INITRAMFS_DIR)/scripts/local-premount/bcachefs
 install: bcachefs $(optional_install)
-	$(INSTALL) -m0755 -D target/release/bcachefs -t $(DESTDIR)$(ROOT_SBINDIR)
+	$(INSTALL) -m0755 -D $(CARGO_TARGET_DIR)/release/bcachefs -t $(DESTDIR)$(ROOT_SBINDIR)
 	$(INSTALL) -m0644 -D bcachefs.8    -t $(DESTDIR)$(PREFIX)/share/man/man8/
 	$(INSTALL) -m0755 -D initramfs/script $(DESTDIR)$(INITRAMFS_SCRIPT)
 	$(INSTALL) -m0755 -D initramfs/hook   $(DESTDIR)$(INITRAMFS_HOOK)
